experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,541 @@
+		// deltas[i] > int(thresholds[i])
+
+		if touched {
+			if d.inputStatuses[i] == PressedStatus {
+				if d.RetriggerOnHold {
+					d.inputStatuses[i] = HeldStatus
+				}
+				continue
+			}
+			d.inputStatuses[i] = PressedStatus
+		} else {
+			d.inputStatuses[i] = OffStatus
+		}
+	}
+
+	return d.inputStatuses, nil
+}
+
+// LinkLEDs link the behavior of the LEDs to the touch sensors.
+// Doing so, disabled the option for the host to set specific LEDs on/off.
+func (d *Dev) LinkLEDs() error {
+	if err := d.regWrapper.WriteUint8(reg_LEDLinking, 0xff); err != nil {
+		return wrap(fmt.Errorf("failed to link LEDs - %s", err))
+	}
+	d.LinkedLEDs = true
+	return nil
+}
+
+// UnlinkLEDs disassociate the LEDs from the input sensors allowing the host to
+// control the LEDs.
+func (d *Dev) UnlinkLEDs() error {
+	if err := d.regWrapper.WriteUint8(reg_LEDLinking, 0x00); err != nil {
+		return wrap(fmt.Errorf("failed to unlink LEDs - %s", err))
+	}
+	d.LinkedLEDs = false
+	return nil
+}
+
+// AllLEDsOn turns all the LEDs on.
+//
+// This is quite more efficient than looping through each led and turn them on.
+func (d *Dev) AllLEDsOn() error {
+	if d.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	return d.regWrapper.WriteUint8(reg_LEDOutputControl, 0xff)
+}
+
+// AllLEDsOff turns all the LEDs off.
+// This is quite more efficient than looping through each led and turn them off.
+func (d *Dev) AllLEDsOff() error {
+	if d.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	return d.regWrapper.WriteUint8(reg_LEDOutputControl, 0x00)
+}
+
+// SetLED sets the state of a LED as on or off
+// Only works if the LEDs are not linked to the sensors
+func (d *Dev) SetLED(idx int, state bool) error {
+	if d.LinkedLEDs {
+		return wrap(fmt.Errorf("can't manually set LEDs when they are linked to sensors"))
+	}
+	if idx > 7 || idx < 0 {
+		return wrap(fmt.Errorf("invalid led idx %d", idx))
+	}
+	if d.Debug {
+		log.Printf("Set LED state %d - %t\n", idx, state)
+	}
+	if state {
+		return d.setBit(reg_LEDOutputControl, idx)
+	}
+	return d.clearBit(reg_LEDOutputControl, idx)
+}
+
+// Reset issues a soft reset to the device using the reset pin
+// if available.
+func (d *Dev) Reset() (err error) {
+	d.ClearInterrupt()
+	if d != nil && d != nil && d.ResetPin != nil {
+		if d.Debug {
+			fmt.Println("cap1188: Resetting the device using the reset pin")
+		}
+		if err = d.ResetPin.Out(gpio.Low); err != nil {
+			return err
+		}
+		if err = d.ResetPin.Out(gpio.High); err != nil {
+			return err
+		}
+		time.Sleep(10 * time.Millisecond)
+		if err = d.ResetPin.Out(gpio.Low); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ClearInterrupt resets the interrupt flag
+func (d *Dev) ClearInterrupt() error {
+	// clear the main control bit
+	return d.clearBit(0x0, 0)
+}
+
+// Standby turns off the capacitive touch sensor inputs. The status registers will
```

Comment:
Let's not export the functions not implemented.

In practice, do we need the 3 followings?

Reply:
in practice we don't need them, they'd be nice to have but I personally don't need them.